openstack/networking/v2/subnets/requests.go

```diff
@@ -88,7 +88,16 @@ type CreateOpts struct {
 
 // ToSubnetCreateMap casts a CreateOpts struct to a map.
 func (opts CreateOpts) ToSubnetCreateMap() (map[string]interface{}, error) {
-	return gophercloud.BuildRequestBody(opts, "subnet")
+	b, err := gophercloud.BuildRequestBody(opts, "subnet")
+	if err != nil {
+		return nil, err
+	}
+
+	if m := b["subnet"].(map[string]interface{}); m["gateway_ip"] == "" {
+		m["gateway_ip"] = nil
```

Comment:
@jtopjian / @jrperritt Shouldn't this line have been `b["gateway_ip"] = nil`?

Else how does `b["gateway_ip"]` get set to `nil`?